### PR TITLE
[RF] RemoveTR and Despike

### PR DESCRIPTION
**Pull request by mattcieslak**
This PR refactors and tests the `RemoveTR` interface and the `Despike` interface. 

For the `RemoveTR` interface
 - [x] Ensure that the output tsv is a proper fmriprep format confounds tsv
 - [x] Ensure the output tsv is indeed truncated 
 - [x] Ensure that both nifti and cifti are correctly truncated
 - [x] Rename interface inputs and outputs to be more clear
 - [x] Check whether we need wb_command to cut times off the beginning of a cifti file

### CODE REVIEW

File: xcp_d/interfaces/prepostcleaning.py [hunk 1]
```diff
@@ -15,12 +15,14 @@ class _removeTRInputSpec(BaseInterfaceInputSpec):
                      mandatory=True,
                      desc=" either bold or nifti ")
     mask_file = File(exists=False, mandatory=False, desc="required for nifti")
-    volumes_to_drop = traits.Int(mandatory=True,
-                                 desc="number of volumes to drop from the beginning")
+    time_todrop = traits.Float(mandatory=True,
```
Comment (mattcieslak) [inline xcp_d/interfaces/prepostcleaning.py:18]: I think we should actually keep this in an integer volumes to drop. We'll calculate this based on the TR in the main code

File: xcp_d/interfaces/prepostcleaning.py [hunk 2]
```diff
@@ -15,12 +15,14 @@ class _removeTRInputSpec(BaseInterfaceInputSpec):
...
+                               desc="number of seconds to drop from the beginning")
     fmriprep_conf = File(
         exists=True,
         mandatory=False,
         desc="confound selected from fmriprep confound matrix")
+    TR = traits.Float(mandatory=True,
```
Comment (mattcieslak) [inline xcp_d/interfaces/prepostcleaning.py:24]: If we're directly using number of volumes to drop we won't need the TR

File: xcp_d/interfaces/prepostcleaning.py [hunk 3]
```diff
@@ -51,9 +53,9 @@ class removeTR(SimpleInterface):
     output_spec = _removeTROutputSpec
 
     def _run_interface(self, runtime):
-
+        volumes_to_drop = np.ceil(self.inputs.time_todrop / self.inputs.TR)
```
Comment (mattcieslak) [inline xcp_d/interfaces/prepostcleaning.py:56]: Let's keep this outside of this function so we don't calculate it twice. We have to calculate it for the boilerplate in the init_postprocess_wf

File: xcp_d/interfaces/prepostcleaning.py [hunk 4]
```diff
@@ -76,15 +78,15 @@ def _run_interface(self, runtime):
...
+            dropped_data = data[volumes_to_drop:]
             dropped_image = nb.Cifti2Image(
                 dropped_data,
                 header=bold_image.header,
                 nifti_header=bold_image.nifti_header)
 
         # It's a Nifti
         else:
-            dropped_data = data[..., self.inputs.volumes_to_drop:]
+            dropped_data = data[volumes_to_drop:]
```
[the remark this answers is not in the document]
Comment (kahinimehta) [reply xcp_d/interfaces/prepostcleaning.py:89]: ahhh

**Pull request merged**